MonoDevelop.FSharpBinding/FSharpSymbolHelper.fs

```diff
@@ -357,11 +364,125 @@ module SymbolTooltips =
         if unionCase.UnionCaseFields.Count > 0 then
            let typeList =
               unionCase.UnionCaseFields
-              |> Seq.map (fun unionField -> unionField.Name ++ asType Symbol ":" ++ asType UserType (escapeText (unionField.FieldType.Format displayContext)))
-              |> String.concat (asType Symbol " * " )
-           unionCase.Name ++ asType Keyword "of" ++ typeList
+              |> Seq.map (fun unionField -> unionField.Name ++ asSymbol ":" ++ asUserType (escapeText (unionField.FieldType.Format displayContext)))
+              |> String.concat (asSymbol " * " )
+           unionCase.Name ++ asKeyword "of" ++ typeList
          else unionCase.Name
 
+    let formatGenericParameter displayContext (param:FSharpGenericParameter) =
+        let chopStringTo (s:string) (c:char) =
+            // chopStringTo "abcdef" 'c' --> "def"
+            if s.IndexOf c <> -1 then
+                let i =  s.IndexOf c + 1
+                s.Substring(i, s.Length - i)
+            else
+                s
+    
+        let tryChopPropertyName (s: string) =
+            // member names start with get_ or set_ when the member is a property
+            let s = 
+                if s.StartsWith("get_", System.StringComparison.Ordinal) || 
+                   s.StartsWith("set_", System.StringComparison.Ordinal) 
+                then s 
+                else chopStringTo s '.'
+        
+            if s.Length <= 4 || (let s = s.Substring(0,4) in s <> "get_" && s <> "set_") then
+                None
+            else 
+                Some(s.Substring(4,s.Length - 4))
+
+        let sb = new StringBuilder()
+        let print symbols = symbols |> Seq.iter (Printf.bprintf sb "%s")
+
+        let printParamName() =
+            print [asSymbol (if param.IsSolveAtCompileTime then "^" else "'")]
```

Comment:
A `print { ... }` CE might make sense, I was thinking about this for tooltip cleanup